src/components/Popups/PopupHighlight.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import noop from 'lodash/noop';
+import { FormattedMessage } from 'react-intl';
+import IconHighlight from '../../icons/IconHighlight';
+import messages from './messages';
+import PopupBase from './PopupBase';
+import { DOMRectMini } from '../../store';
+import { Options } from './Popper';
+import './PopupHighlight.scss';
+
+export type Props = {
+    onClick?: (event: React.MouseEvent) => void;
+    rect: DOMRectMini;
+};
+
+const options: Partial<Options> = {
+    modifiers: [
+        {
+            name: 'arrow',
+            options: {
+                element: '.ba-Popup-arrow',
+            },
+        },
+        {
+            name: 'eventListeners',
+            options: {
+                scroll: false,
+            },
+        },
+        {
+            name: 'offset',
+            options: {
+                offset: [0, 8],
+            },
+        },
+        {
+            name: 'preventOverflow',
+            options: {
+                padding: 5,
+            },
+        },
+    ],
+    placement: 'bottom',
+};
+
+export default function PopupCursor({ onClick = noop, rect }: Props): JSX.Element {
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
```

Comment:
Should this be a `callbackRef` instead?

Reply:
`useCallBack` doesn't support cleanup function so we can't remove listeners.

Reply:
Adding event listeners doesn't need a re-render. Discussed offline.